### PR TITLE
put MAP in the sensor model

### PR DESCRIPTION
**Pull request by mck1117**
Required some new functionality to get map averaging to behave itself, but I think this solution is pretty clean!

Both fast and slow MAP use the same conversion function. Slow MAP uses traditional slow ADC.  Fast map uses a special path to update the fast MAP sensor.

#2667

### CODE REVIEW

File: firmware/init/sensor/init_map.cpp [hunk 1]
```diff
@@ -32,13 +22,120 @@ static FunctionPointerSensor baroSensor(SensorType::BarometricPressure,
...
 
+// This converter is shared between both fast and slow: the only difference is
+// how the *voltage* is determined, not how its converted to a pressure.
+static LinearFunc mapConverter;
+static FunctionalSensor slowMapSensor(SensorType::MapSlow, MS2NT(50));
+
+// lowest reasonable idle is maybe 600 rpm
+// one sample per cycle (1 cylinder, or "sample one cyl" mode) gives a period of 100ms
+// add some margin -> 200ms timeout for fast MAP sampling
+static FunctionalSensor fastMapSensor(SensorType::MapFast, MS2NT(200));
```
Comment (andreika-git) [inline firmware/init/sensor/init_map.cpp:33]: Is it correct that the "fast" sensor has larger period than the "slow" one?
Usually frequency = 1 / period

Comment (mck1117) [reply firmware/init/sensor/init_map.cpp:33]: That 200ms period is the timeout - which in this case is how long we wait before falling back to the "slow" sensor.  The slow sensor's still updates at 200hz or whatever, but asynchronously, hence its much shorter timeout.

File: firmware/controllers/sensors/sensor_type.h [hunk 2]
```diff
@@ -77,6 +77,9 @@ enum class SensorType : unsigned char {
 
 	TurbochargerSpeed = 35,
 
+	MapFast = 36,
```
Comment (rusefillc) [inline firmware/controllers/sensors/sensor_type.h:80]: code comment with a quick summary of what is fast MAP vs what is slow MAP

Comment (mck1117) [reply firmware/controllers/sensors/sensor_type.h:80]: I guess - they're just averaged (synchronous) map vs asynchronous map

**Comment by rusefillc** [comment]
Drama at #3325

**Pull request merged**